### ISSUE TITLE
Unable to CONNECT to authority with underscore

### ISSUE THREAD

**Issue by GpMits** [zaproxy/zaproxy#5252]
**Describe the bug**
When an URI with an underscore character ('_') in the authority part, f.e. "ex_ample.something.com" is used with proxy, the CONNECT request, which has no scheme, will make ZAP throw an exception causing an error "ERR_EMPTY_RESPONSE" in the browser. The problem is related to the parse URI method which does not know to handle such URIs 

**To Reproduce**
1. Start ZAP UI or a browser using the ZAP proxy
2. Go to the URL "ex_ample.blogspot.com"
3. See error (browser and log)

**Expected behavior**
No error shows up and the website content is loaded.

**Software versions**
 - ZAP: 2.7.0 (but is also happening using the current development branch)
 - OS: Windows 10
 - Java: openjdk version "1.8.0_202"
 - Browser: chrome 72, ZAP built in browser

**Errors from the zap.log file**
```
55795 [ZAP-ProxyThread-5] ERROR org.parosproxy.paros.network.HttpRequestHeader  - incorrect scheme
org.apache.commons.httpclient.URIException: incorrect scheme
        at org.apache.commons.httpclient.URI.parseUriReference(URI.java:1973)
        at org.apache.commons.httpclient.URI.<init>(URI.java:181)
        at org.parosproxy.paros.network.HttpRequestHeader.parseURI(HttpRequestHeader.java:666)
        at org.parosproxy.paros.network.HttpRequestHeader.parse(HttpRequestHeader.java:451)
        at org.parosproxy.paros.network.HttpRequestHeader.setMessage(HttpRequestHeader.java:275)
        at org.parosproxy.paros.network.HttpRequestHeader.<init>(HttpRequestHeader.java:163)
        at org.parosproxy.paros.network.HttpInputStream.readRequestHeader(HttpInputStream.java:68)
        at org.parosproxy.paros.core.proxy.ProxyThread.run(ProxyThread.java:284)
        at java.lang.Thread.run(Unknown Source)
55795 [ZAP-ProxyThread-6] ERROR org.parosproxy.paros.network.HttpRequestHeader  - incorrect scheme
org.apache.commons.httpclient.URIException: incorrect scheme
        at org.apache.commons.httpclient.URI.parseUriReference(URI.java:1973)
        at org.apache.commons.httpclient.URI.<init>(URI.java:181)
        at org.parosproxy.paros.network.HttpRequestHeader.parseURI(HttpRequestHeader.java:666)
        at org.parosproxy.paros.network.HttpRequestHeader.parse(HttpRequestHeader.java:451)
        at org.parosproxy.paros.network.HttpRequestHeader.setMessage(HttpRequestHeader.java:275)
        at org.parosproxy.paros.network.HttpRequestHeader.<init>(HttpRequestHeader.java:163)
        at org.parosproxy.paros.network.HttpInputStream.readRequestHeader(HttpInputStream.java:68)
        at org.parosproxy.paros.core.proxy.ProxyThread.run(ProxyThread.java:284)
        at java.lang.Thread.run(Unknown Source)
55796 [ZAP-ProxyThread-5] WARN org.parosproxy.paros.core.proxy.ProxyThread  - Malformed Header:
org.parosproxy.paros.network.HttpMalformedHeaderException: incorrect scheme
        at org.parosproxy.paros.network.HttpRequestHeader.setMessage(HttpRequestHeader.java:288)
        at org.parosproxy.paros.network.HttpRequestHeader.<init>(HttpRequestHeader.java:163)
        at org.parosproxy.paros.network.HttpInputStream.readRequestHeader(HttpInputStream.java:68)
        at org.parosproxy.paros.core.proxy.ProxyThread.run(ProxyThread.java:284)
        at java.lang.Thread.run(Unknown Source)
55800 [ZAP-ProxyThread-6] WARN org.parosproxy.paros.core.proxy.ProxyThread  - Malformed Header:
org.parosproxy.paros.network.HttpMalformedHeaderException: incorrect scheme
        at org.parosproxy.paros.network.HttpRequestHeader.setMessage(HttpRequestHeader.java:288)
        at org.parosproxy.paros.network.HttpRequestHeader.<init>(HttpRequestHeader.java:163)
        at org.parosproxy.paros.network.HttpInputStream.readRequestHeader(HttpInputStream.java:68)
        at org.parosproxy.paros.core.proxy.ProxyThread.run(ProxyThread.java:284)
        at java.lang.Thread.run(Unknown Source)
```

**Additional context**
Issue was discussed in this topic: https://groups.google.com/forum/#!topic/zaproxy-develop/hc5XvvykVeU

**Comment by GpMits** [comment]
I'll try to fix this asap

**Comment by thc202** [comment]
Tweaked the title, it's already possible to access URLs with underscore in the host (e.g. `http://a_b.example.com/` works), just not `CONNECT` to them (e.g. when using HTTPS).

**Comment by thc202** [comment]
Set to 2.8.0.

**Comment by thc202** [comment]
Fixed in #5266.
